crates/node-builder/src/components/exex.rs

```diff
@@ -0,0 +1,106 @@
+use std::{
+    collections::VecDeque,
+    future::Future,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+use reth_primitives::BlockNumber;
+use reth_provider::{CanonStateNotification, CanonStateNotifications};
+use tokio::sync::{mpsc::UnboundedSender, watch};
+use tokio_util::sync::PollSender;
+
+// todo
+// - exex height
+// - exex channel size
+// - manager buffer metrics
+// - throughput
+struct ExExMetrics;
+
+/// The execution extension manager.
+///
+/// The manager is responsible for:
+///
+/// - Receiving relevant events from the rest of the node, and sending these to the execution
+/// extensions
+/// - Backpressure
+/// - Error handling
+/// - Monitoring
+///
+/// TBD
+struct ExExManager {
+    // todo: exex receiver - get exex events
+    /// Channels from the manager to execution extensions.
+    exex_tx: Vec<PollSender<CanonStateNotification>>,
+
+    /// [`CanonStateNotification`] channel from the execution stage.
+    exec_rx: (), //UnboundedReceiverStream<CanonStateNotification>,
+    /// [`CanonStateNotification`] channel from the blockchain tree.
+    tree_rx: CanonStateNotifications,
+
+    /// Internal buffer of [`CanonStateNotification`]s.
+    buffer: VecDeque<CanonStateNotification>,
+    /// Max internal buffer size.
+    max_capacity: usize,
+    /// Current buffer capacity.
+    ///
+    /// Used to inform the execution stage of possible batch sizes.
+    current_capacity: Arc<AtomicUsize>,
+
+    /// Whether the manager is ready to receive new notifications.
+    is_ready: watch::Sender<bool>,
+
+    /// block number for pruner/exec stage (tbd)
+    /// todo: this is inclusive, note that in exex too, maybe rename FinishedHeight
+    block: watch::Sender<BlockNumber>,
```

Comment:
let's move the handle type in here, this way a new handle can be obtained from the manager right before we spawn it

Reply:
actually, are we sure we want that? that would mean that we always have some senders/receivers open

maybe thats ok?

Reply:
this just ensures the channel is always open, we use that a bunch, for example

https://github.com/paradigmxyz/reth/blob/bdca72273c16c526d885a172732a33cd02d29746/crates/net/network/src/transactions/mod.rs#L226-L234

https://github.com/paradigmxyz/reth/blob/bdca72273c16c526d885a172732a33cd02d29746/crates/net/network/src/transactions/mod.rs#L309-L312